templates/partials/summary/summary.html

```diff
@@ -6,30 +6,58 @@
   {% set summaryGroups = [] %}
 
   {%- for group in content.summary.groups if group.blocks -%}
-    {% do summaryGroups.append (
-      {
+
+      {% set links = {'add_link_text': "", "add_link": "", "empty_link_text": ""} %}
+
+      {% for block in group.blocks %}
+        {% if block.type == "List" %}
+          {% do links.update({'add_link_text': block.add_link_text}) %}
+          {% do links.update({'add_link': block.add_link}) %}
+          {% do links.update({'empty_list_text': block.empty_list_text}) %}
+        {% endif %}
+      {% endfor %}
```

Comment:
Instead of looping over every block to build the link, given you are passing a list collector built context, pass in a built link definition as part of group. Then you can just pass it directly via `group.links`

Reply:
The loop is fixed now:
```
{% if group.links %}
    {% set links = {"add_link_text": group.links.add_link_text, "add_link": group.links.add_link, "empty_link_text": group.links.empty_list_text} %}
{% else %}
    {% set links = {"add_link_text": "",  "add_link": "", "empty_link_text": ""} %}
{% endif %}
```
Some tweaks were needed in blocks generation in `Group` class.